src/edu/ucsd/library/xdre/statistic/analyzer/DAMStatistic.java

```diff
@@ -349,11 +345,22 @@ public void export(Connection con) throws SQLException{
 		
 		//STATS_DLP_OBJECT_ACCESS_INSERT insert
 		try{
-			// eliminate the counts from curator access with redirected
+			// eliminate the counts from curator access with redirects from public access
 			for (String key : itemsMapPrivate.keySet()) {
 				if (itemsMap.containsKey(key)) {
-					int pubAccess = itemsMap.get(key).getView() - itemsMapPrivate.get(key).getView();
-					itemsMap.get(key).setView(pubAccess);
+					StatsObjectAccess pubObjAccess = itemsMap.get(key);
+					StatsObjectAccess privObjAccess = itemsMapPrivate.get(key);
+					for (String ip : privObjAccess.getObjectCounters().keySet()) {
+						ObjectCounter pubCounter = pubObjAccess.getCounter(ip);
+						ObjectCounter privCounter = privObjAccess.getCounter(ip);
```

Comment:
what happens if `privCounter` is null? Could that happen?

Reply:
@mcritchlow Theoretically the `privCounter` won't be null because a new instance of ObjectCounter is created and `put` into the map with the ip as the key at the same time. So when looping through the `privCounter` elements in the map, we won't get any null elements that trigger the NullPointerException error. But I think it won't harm to do a safe check. Please let me know if you still prefer to do that safe check. Thanks.

Reply:
Ah ok, no that's probably fine. Just wanted to make sure there wasn't a possible edge case that we might be missing there.